### PR TITLE
Refactor the command palette and add new features for create

### PR DESCRIPTION
**Pull request by OliverMKing**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)
Feature # (details)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Is it a  breaking change which will impact consuming tool(s).

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/commands/runDraftTool/runDraftDockerfile.ts [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+export async function multiStepInput(context: ExtensionContext, destination: string) {
+    const title = 'Draft a Dockerfile from source code';
+    const languages = ['clojure', 'c#', 'erlang', 'go', 'gomodule', "java", "gradle", "javascript", "php", "python", "rust", "swift"];
+	const languageLabels: QuickPickItem[] = languages.map(label => ({ label }));
+
+	interface State {
+		title: string;
+		step: number;
+		totalSteps: number;
+        language: string;
```
Comment (davidgamero) [inline src/commands/runDraftTool/runDraftDockerfile.ts:34]: indentation

File: src/commands/runDraftTool/runDraftDockerfile.ts [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+    const totalSteps = 5;
+    async function inputSourceCodeFolder(input: MultiStepInput, state: Partial<State>, step: number) {
+		state.sourceFolder = await input.showInputBox({
+			title,
+			step: step,
+			totalSteps: totalSteps,
+			value: typeof state.sourceFolder === 'string' ? state.sourceFolder : '',
+			prompt: 'Folder with your source code e.g. ./src)',
+            validate: async (file: string) => {
```
Comment (davidgamero) [inline src/commands/runDraftTool/runDraftDockerfile.ts:56]: indentation and formatting

File: src/commands/runDraftTool/runDraftDockerfile.ts [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+            const topLanguage = Object.keys(results).reduce((prev, key) => {
+                if (prev === "") return key
+
+                const prevVal = results[prev].bytes;
+                const currVal = results[key].bytes;
+                return prevVal > currVal ? prev : key;
+            }, "");
+
+            // convert to expected language form
+            const languages = ['clojure', 'c#', 'erlang', 'go', 'gomodule', "java", "gradle", "javascript", "php", "python", "rust", "swift"];
```
Comment (davidgamero) [inline src/commands/runDraftTool/runDraftDockerfile.ts:102]: is this a repeat of the languages const higher up in this file?

**Review by davidgamero — CHANGES_REQUESTED** [review]

**Comment by davidgamero** [comment]
we should definitely add a linter/formatter to this. It's not easy to read with inconsistent indentation

**Comment by OliverMKing** [comment]
> we should definitely add a linter/formatter to this project. It's not easy to read with inconsistent indentation

agreed and i've brought it up but we can't right now. the large merge back to main will be too hard to read

**Review by davidgamero — APPROVED** [review]

**Pull request merged**